File: java/src/tk/csparpa/gofdp/factorymethod/ConcreteProduct.java
package tk.csparpa.gofdp.factorymethod;

/**
 * 
 * Concrete implementation of products interface. Modify it in order to 
 * instantiate your own ones!
 * 
 */
public class ConcreteProduct implements AbstractProduct {
	
	public ConcreteProduct(){
		System.out.println("Created: concrete product instance");		
	}
	
	@Override
	public void feature() {
		System.out.println("Called: concrete product feature");
	}
}

File: python/factorymethod/concrete_product.py
from abstract_product import AbstractProduct

class ConcreteProduct(AbstractProduct):

    """
    Concrete implementation of products interface. Modify it in order to 
    instantiate your own ones!
    """

    def __init__(self):
        print("Created: concrete product instance")        
    
    def feature(self):
        print("Called: concrete product feature")
